cilium-cli/connectivity/tests/pod.go

```diff
@@ -253,3 +256,167 @@ func (s *podToPodNoFrag) Run(ctx context.Context, t *check.Test) {
+	}
+
+	// Delete echo pods entries from ipcache
+	for _, echo := range ct.EchoPods() {
+		echoIP := net.ParseIP(echo.Address(features.IPFamilyV4))
+		for _, ipcache := range ipcaches {
+			ipcacheKey := strings.Split(fmt.Sprintf("0x40 0x00 0x00 0x00 0x00 0x00 0x00 0x01 0x%02x 0x%02x 0x%02x 0x%02x 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00 0x00", echoIP[12], echoIP[13], echoIP[14], echoIP[15]), " ")
+			lookupCmd := append(strings.Split(fmt.Sprintf("bpftool --json map lookup id %d key hex", ipcache.ID), " "), ipcacheKey...)
+			updateCmd := append(strings.Split(fmt.Sprintf("bpftool map update id %d key hex", ipcache.ID), " "), ipcacheKey...)
+			deleteCmd := append(strings.Split(fmt.Sprintf("bpftool map delete id %d key hex", ipcache.ID), " "), ipcacheKey...)
+
+			output, err := ciliumPod.K8sClient.ExecInPod(ctx, ciliumPod.Pod.Namespace, ciliumPod.Pod.Name, defaults.AgentContainerName, lookupCmd)
+			if err != nil {
+				ct.Warnf(`failed to lookup IP cache entry: "%s", %v, "%s"`, lookupCmd, err, output.String())
+				continue
+			}
+			lookup := bpfMapLookup{}
+			if err = json.Unmarshal(output.Bytes(), &lookup); err != nil {
+				ct.Warnf(`failed to unmarshal output "%s": %v`, output.String(), err)
+				continue
+			}
+			if strings.Join(lookup.Key, " ") != strings.Join(ipcacheKey, " ") {
+				ct.Debugf("ipcache key not found: %s", strings.Join(ipcacheKey, " "))
+				continue
+			}
+
+			if output, err = ciliumPod.K8sClient.ExecInPod(ctx, ciliumPod.Pod.Namespace, ciliumPod.Pod.Name, defaults.AgentContainerName, deleteCmd); err != nil {
+				ct.Warnf(`failed to delete IP cache entry: "%s", %v, "%s"`, deleteCmd, err, output.String())
+				continue
+			}
+
+			defer func(updateCmd []string, lookup bpfMapLookup) {
+				updateCmd = append(updateCmd, "value", "hex")
+				updateCmd = append(updateCmd, lookup.Value...)
+				output, err := ciliumPod.K8sClient.ExecInPod(ctx, ciliumPod.Pod.Namespace, ciliumPod.Pod.Name, defaults.AgentContainerName, updateCmd)
+				if err != nil {
+					ct.Warnf(`failed to restore IP cache entry: "%s", %v, "%s"`, updateCmd, err, output.String())
+				}
+			}(updateCmd, lookup)
+		}
+	}
+
+	unencryptedCountCmd := []string{
+		"/bin/sh", "-c",
+		`cilium metrics list -o json | jq '.[] | select((.name == "cilium_drop_count_total") and (.labels.reason | IN("Traffic is unencrypted"))) | .value'`,
+	}
+	prevUnencryptedCount := 0
+	for _, ciliumPod := range ct.CiliumPods() {
+		output, err = ciliumPod.K8sClient.ExecInPod(ctx, ciliumPod.Pod.Namespace, ciliumPod.Pod.Name, defaults.AgentContainerName, unencryptedCountCmd)
+		if err != nil {
+			ct.Fatalf(`failed to execute "%s" in %s: %v`, unencryptedCountCmd, ciliumPod.Pod.Name, err)
+		}
+		if output.String() != "" {
+			count, err := strconv.Atoi(strings.TrimSpace(output.String()))
+			if err != nil {
+				ct.Fatalf(`failed to parse "%s": %v`, output.String, err)
+			}
+			prevUnencryptedCount += count
+		}
+	}
```

Comment:
nit: Move this into a helper function for better readability (since it seems to be used twice)